### ISSUE TITLE
CommError: The comm is not connected when changing Matplotlib backend

### ISSUE THREAD

**Issue by WisamAbbasi** [spyder-ide/spyder#15875]
### What steps will reproduce the problem?

when choosing inline backend graphics, the spyder returns an error message

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\preferences\widgets\configdialog.py", line 127, in accept
    configpage.apply_changes()
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\preferences\api.py", line 110, in apply_changes
    self.apply_callback()
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\preferences\api.py", line 142, in <lambda>
    self._apply_settings_tabs(self.changed_options))
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\preferences\api.py", line 170, in _apply_settings_tabs
    self.apply_settings(options)
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\api\preferences.py", line 104, in wrapper
    func(opts)
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 447, in apply_plugin_settings
    for client in self.clients]
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 447, in <listcomp>
    for client in self.clients]
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 343, in get_matplotlib_backend
    blocking=True).get_matplotlib_backend()
  File "C:\Users\wisam\anaconda3\envs\PEA Tabular Project\lib\site-packages\spyder_kernels\comms\commbase.py", line 550, in __call__
    raise CommError("The comm is not connected.")
spyder_kernels.comms.commbase.CommError: The comm is not connected.
```

## Versions

* Spyder version: 5.0.0 
* Python version: 3.6.13
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  3.0.4 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.16.1 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  22.3.0 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.8.1 (OK)
pylint >=1.0                  :  2.7.4 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2           :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=0.5.7             :  1.0.2 (OK)
qtconsole >=5.0.3             :  5.0.3 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.8.3                 :  0.9.4 (OK)
setuptools >=39.0.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  3.5.3 (OK)
spyder_kernels >=2.0.1,<2.1.0 :  2.0.1 (NOK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.19.5 (OK)
pandas >=1.1.1                :  1.1.5 (OK)
scipy >=0.17.0                :  1.5.2 (OK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by steff456** [comment]
Hi @WisamAbbasi,

This is probably an issue with your firewall and/or antivirus. Please try to deactivate them and try again. Also if you want you can try to upgrade to the latest version, 5.0.4 and check if the error persists as well.

**Comment by WisamAbbasi** [comment]
I tried to deactivate them, but not working either and I am using the spyder v5

**Comment by steff456** [comment]
Are you using the latest release 5.0.4 or 5.0.0?

**Comment by WisamAbbasi** [comment]
It is 5.0.4

**Comment by ccordoba12** [comment]
@WisamAbbasi, is the variable explorer working for you?

**Comment by ccordoba12** [comment]
@dalthviz, could you check if you can reproduce this error with our Windows installer (see issue #16272) and Anaconda?

**Comment by ccordoba12** [comment]
@impact27, do you have an idea about what's happening here?

**Comment by dalthviz** [comment]
@ccordoba12 I was unable to reproduce this (is working for me with the installer and anaconda/conda-forge).

**Comment by Walberf17** [comment]
I can't initialize any other thing because there is no connection on
kernel, so I can't do anything.

Em ter, 24 de ago de 2021 12:24, Carlos Cordoba ***@***.***>
escreveu:

**Comment by ccordoba12** [comment]
Ok, that's expected. Then this problem is probably caused by your firewall or antivirus. Please deactivate them and try again.

**Comment by Walberf17** [comment]
Sorry for the late answer, I tried what you suggested, but no results...

Em qua, 25 de ago de 2021 01:07, Carlos Cordoba ***@***.***>
escreveu:

**Comment by ccordoba12** [comment]
Sorry, I don't have more advice for you. But this definitely not a problem in Spyder.

**Comment by Walberf17** [comment]
Ok, thanks.

Em qua, 25 de ago de 2021 10:45, Carlos Cordoba ***@***.***>
escreveu:

**Comment by ccordoba12** [comment]
I found this is related to a bug when setting interactive Matplotlib backends. It'll be fixed in the next version of spyder-kernels (2.1.3), to be released soon.

Spyder itself will pick that change by default in its next version (5.2.0), to be released later this month.
